src/main/java/uk/gov/justice/json/generator/JsonArrayGenerator.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.justice.json.generator;
+
+import static javax.json.Json.createArrayBuilder;
+
+import uk.gov.justice.services.test.utils.core.random.Generator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.json.JsonArray;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonValue;
+
+import org.everit.json.schema.ArraySchema;
+import org.everit.json.schema.EmptySchema;
+import org.everit.json.schema.Schema;
+
+/**
+ * Random generator for JSON arrays.
+ */
+public class JsonArrayGenerator extends Generator<JsonArray> {
+
+    public static final int DEFAULT_MIN_ITEMS = 0;
+    public static final int DEFAULT_MAX_ITEMS = 10;
+
+    private List<Generator<JsonValue>> itemGenerators;
+    private Generator<JsonValue> itemGenerator;
+
+    private int minItems;
+    private int maxItems;
+
+    /**
+     * TODO: Unique values
+     * Constructor.
+     * @param schema the array schema to base the generator on
+     */
+    public JsonArrayGenerator(final ArraySchema schema) {
+
+        minItems = schema.getMinItems() != null ? schema.getMinItems() : DEFAULT_MIN_ITEMS;
+        maxItems = schema.getMaxItems() != null ? schema.getMaxItems() : DEFAULT_MAX_ITEMS;
+
+        if (schema.getAllItemSchema() != null && schema.getItemSchemas() == null) {
+            initialiseForListArrays(schema.getAllItemSchema());
+        } else if (schema.getItemSchemas() != null && schema.getAllItemSchema() == null) {
+            initialiseForTupleArrays(
+                    schema.getItemSchemas(),
+                    schema.permitsAdditionalItems(),
+                    Optional.ofNullable(schema.getSchemaOfAdditionalItems()));
+        } else if (schema.getItemSchemas() == null && schema.getAllItemSchema() == null) {
+            initialiseForUnspecifiedArrays();
+        } else {
+            throw new JsonGenerationException("Inconsistent array schema");
+        }
+    }
```

Comment:
Is this fully tested? as in every path and every boolean condition? It's rather hard to see from this review code